### ISSUE TITLE
自己的数据转成tusimple后 训练的时候出现loss=nan的情况

### ISSUE THREAD

**Issue by walkmanyang** [cfzd/Ultra-Fast-Lane-Detection-v2#5]
你好，很感谢无私你的分享！
我想请问一下 我这边使用自己数据转成tusimple格式数据集进行训练 ， 车道线的点我自己在原图上画出来也是没问题，但是训练途中在读取十几张图片后 loss变成了nan，这是什么原因造成的啊
期望得到你的回复，谢谢！！！

**Comment by cfzd** [comment]
@walkmanyang 
你可以尝试降低学习率

**Comment by walkmanyang** [comment]
非常感谢，已经work了！

**Comment by zhenghao222** [comment]
你好，在训练过程中你的Loss与top大约稳定在多少呢

**Comment by walkmanyang** [comment]
> 你好，在训练过程中你的Loss与top大约稳定在多少呢

好久之前弄得了 ， 忘记了， 你可以训练完后可以测试一下你的数据，如果效果可以，不看参数也行

**Comment by dtdzsf** [comment]
请问如何将自己的数据集转换为tusimple格式的数据集呢

**Comment by zhenghao222** [comment]
> 请问如何将自己的数据集转换为tusimple格式的数据集呢

你可以使用与v1相同的标注方法，例如labelme，生成车道线的二值图像，但v2还需要生成tusimple_anno_cache.json

**Comment by dtdzsf** [comment]
> > 请问如何将自己的数据集转换为tusimple格式的数据集呢
> 
> 你可以使用与v1相同的标注方法，例如labelme，生成车道线的二值图像，但v2还需要生成tusimple_anno_cache.json

是的 我在想办法如何生成 tusimple_anno_cache.json  您由什么好的办法吗

**Comment by zhenghao222** [comment]
> > > 请问如何将自己的数据集转换为tusimple格式的数据集呢
> > 
> > 
> > 你可以使用与v1相同的标注方法，例如labelme，生成车道线的二值图像，但v2还需要生成tusimple_anno_cache.json
> 
> 是的 我在想办法如何生成 tusimple_anno_cache.json 您有什么好的办法吗

可以从你生成的图像出发，或者对车道线的点进行直线拟合寻找或插值处该行点的坐标

**Comment by dtdzsf** [comment]
> > > > 请问如何将自己的数据集转换为tusimple格式的数据集呢
> > > 
> > > 
> > > 你可以使用与v1相同的标注方法，例如labelme，生成车道线的二值图像，但v2还需要生成tusimple_anno_cache.json
> > 
> > 
> > 是的 我在想办法如何生成 tusimple_anno_cache.json 您有什么好的办法吗
> 
> 可以从你生成的图像出发，或者对车道线的点进行直线拟合寻找或插值处该行点的坐标

我尝试进行对数据进行转换，进行训练的时候遇到loss = nan的问题  ，降低了学习率 也没办法解决 可能我的生成程序有问题  可以看一下你的生成程序吗？

**Comment by zhenghao222** [comment]
all_points = np.zeros((4, 56, 2), dtype=np.float)
                    the_anno_row_anchor = np.array([160, 170, 180, 190, 200, 210, 220, 230, 240, 250, 260,
                                                    270, 280, 290, 300, 310, 320, 330, 340, 350, 360, 370,
                                                    380, 390, 400, 410, 420, 430, 440, 450, 460, 470, 480,
                                                    490, 500, 510, 520, 530, 540, 550, 560, 570, 580, 590,
                                                    600, 610, 620, 630, 640, 650, 660, 670, 680, 690, 700,
                                                    710])
                    all_points[:, :, 1] = np.tile(the_anno_row_anchor, (4, 1))
                    all_points[:, :, 0] = -99999
                                   for shape in data['shapes']:
                                lane_name = shape['label']
                                lane_point = shape['points']
                                # print(lane_name)
                                # print(lane_point)
                                lane_x = []
                                lane_y = []
                                for point in lane_point:
                                    lane_x.append(point[0])
                                    lane_y.append(point[1])

                                intersection_x,intersection_y = lane_point_fitting(lane_y,lane_x,lane_name,the_anno_row_anchor)
                                xx = np.array(intersection_x)
                                yy = ((np.array(intersection_y) - 160) / 10).astype(int)

                                if lane_name == 'lane2':
                                    all_points[1, yy, 0] = xx
                                    draw(label, lane_x,lane_y, 2,show=True)
                                elif lane_name == 'lane1':
                                    all_points[0, yy, 0] = xx
                                    draw(label, lane_x,lane_y, 1,show=True)

                                elif lane_name == 'lane3':
                                    all_points[2, yy, 0] = xx
                                    draw(label, lane_x,lane_y, 3,show=True)

                                elif lane_name == 'lane4':
                                    all_points[3, yy, 0] = xx
                                    draw(label, lane_x,lane_y, 4,show=True)
仅贴出部分需求的重点程序供参考

**Comment by dtdzsf** [comment]
![31667450153_ pic](https://user-images.githubusercontent.com/82217009/199648421-6bd4fd23-d04f-41d2-b0ef-11aaf109b30b.jpg)
请问下这个是什么原因

**Comment by liwankuang** [comment]
> 非常感谢，已经work了！

楼主，损失为nan最后咋解决了？一直降低学习率，也没效果
